### PR TITLE
Bw 735

### PR DESCRIPTION
**Pull request by milansar**

### CODE REVIEW

**Comment by milansar** [comment]
@anihalaney , PR is ready for your review

File: projects/trivia/src/app/game-play/components/game-dialog/game-dialog.ts [hunk 1]
```diff
@@ -309,28 +323,45 @@ export class GameDialog {
...
-              () => {
-                // load question screen timer
-                this.utils.unsubscribe([this.timerSub]);
-                this.showBadge = false;
-                this.subscribeQuestion();
-                this.cd.detectChanges();
-              }
-            );
+          if (this.game.gameOptions.isBadgeWithCategory) {
+              this.questionSub = this.gameQuestionObs.subscribe(question => {
```
Comment (anihalaney) [inline projects/trivia/src/app/game-play/components/game-dialog/game-dialog.ts]: is this being unsubscribed?

File: projects/trivia/src/app/game-play/components/game-dialog/game-dialog.ts [hunk 2]
```diff
@@ -309,28 +323,45 @@ export class GameDialog {
...
+          }
+        }
+      );
+  }
+
+  showNextBadgeToBeWon(question?: Question) {
+    this.showLoader = false;
+    this.showBadge = true;
+    this.timer = this.MAX_TIME_IN_SECONDS_BADGE;
+    this.timerSub = timer(
```
Comment (anihalaney) [inline projects/trivia/src/app/game-play/components/game-dialog/game-dialog.ts]: is this being unsubscribed?

Comment (piya99) [reply projects/trivia/src/app/game-play/components/game-dialog/game-dialog.ts]: yes

File: projects/trivia/src/app/game-play/components/game-dialog/game-dialog.ts (hunk 1 above)
Comment (piya99) [reply projects/trivia/src/app/game-play/components/game-dialog/game-dialog.ts]: yes

**Pull request merged**